offixsdk/src/main/java/org/aerogear/offixsdk/ClientGenerator.kt

```diff
@@ -0,0 +1,139 @@
+        } ?: kotlin.run {
+            apClient = ApolloClient.builder()
+                .okHttpClient(getOkhttpClient()!!)
+                .normalizedCache(cacheFactory, cacheResolver())
+                .subscriptionTransportFactory(
+                    WebSocketSubscriptionTransport.Factory(
+                        serverUrl,
+                        getOkhttpClient()!!
+                    )
+                )
+                .serverUrl(serverUrl)
+//              .defaultHttpCachePolicy(HttpCachePolicy.CACHE_FIRST)
+                .build()
+        }
+
+        return apClient
+    }
+
+    private fun getOkhttpClient(): OkHttpClient? {
+
+        //Adding HttpLoggingInterceptor() to see the response body and the results.
+        httpClient?.let {
+            return it
+        } ?: kotlin.run {
+            httpClient = OkHttpClient.Builder()
+                .addInterceptor(LoggingInterceptor())
+                .addInterceptor(getResponseInterceptor(applicationContext)!!)
+                .build()
+        }
+        return httpClient
+    }
+
+    //function to get the response after query/mutation is performed, can get conflict messages and so on.
+    private fun getResponseInterceptor(context: Context): Interceptor? {
+
+        conflictInterceptor?.let {
+            return it
+        } ?: kotlin.run {
+            conflictInterceptor = Interceptor {
+                val request = it.request()
+
+                //all the HTTP work happens, producing a response to satisfy the request.
+                val response = it.proceed(request)
+
+                //https://stackoverflow.com/a/33862068/10189663
+
+                val responseBody = response.body()
+                val bufferedSource = responseBody?.source()
+                bufferedSource?.request(Long.MAX_VALUE)
+                val buffer = bufferedSource?.buffer()
+                val responseBodyString = buffer?.clone()?.readString(Charset.forName("UTF-8")) ?: ""
+
+                Log.e("UtillClass", " Interceptor : $responseBodyString")
+
+                //To see for conflict, "VoyagerConflict" which comes in the message is searched for.
+                if (responseBodyString.contains("VoyagerConflict")) {
+                    Log.e(TAG, " Conflcit in offixClient")
+                }
+                return@Interceptor response
+            }
+        }
+        return conflictInterceptor
+    }
+
+    private fun cacheResolver(): CacheKeyResolver {
+        return object : CacheKeyResolver() {
+            override fun fromFieldRecordSet(field: ResponseField, recordSet: Map<String, Any>): CacheKey {
+                Log.e(TAG, "fromFieldRecordSet ${(recordSet["id"] as String)}")
+//                if (recordSet.containsKey("id")) {
+//                    val typeNameAndIDKey = recordSet["__typename"].toString() + "." + recordSet["id"]
+//                    return CacheKey.from(typeNameAndIDKey)
+//                }
+//                return CacheKey.NO_KEY
```

Comment:
Commented out code actually was right. The one above will fail badly when id is not found (but we do not have this in our example app)